src/utils/protocol/downloadProtocol.js

```diff
@@ -23,27 +27,37 @@ const downloadProtocol = inEnvironment((environment) => {
     return (uri) => {
       const tempPath = (electron.app || electron.remote.app).getPath('temp');
 
-      return getProtocolNameFromUri(uri)
-        .then(protocolName => path.join(tempPath, protocolName))
-        .then(destination => writeStream(destination, request(uri)));
+      return getURL(uri)
+        .then(url => [
+          url.href,
+          path.join(tempPath, getProtocolNameFromUrl(url)),
+        ])
+        .catch(friendlyErrorMessage("The location you gave us doesn't seem to be valid. Check the location, and try again."))
+        .then(([url, destination]) => writeStream(destination, request({ uri: url })))
+        .catch(friendlyErrorMessage("Your device doesn't have an active internet connection, so we weren't able to fetch your protocol at this time. Connect to a network, and try again."));
     };
   }
 
   if (environment === environments.CORDOVA) {
     return uri =>
-      getProtocolNameFromUri(uri)
-        .then(protocolName => `cdvfile://localhost/temporary/${protocolName}`)
-        .then(destination =>
+      getURL(uri)
+        .then(url => [
+          url.href,
+          `cdvfile://localhost/temporary/${getProtocolNameFromUrl(url)}`,
+        ])
+        .catch(friendlyErrorMessage("The location you gave us doesn't seem to be valid. Check the location, and try again."))
+        .then(([url, destination]) =>
           new Promise((resolve, reject) => {
             const fileTransfer = new FileTransfer();
             fileTransfer.download(
-              encodeURI(uri),
+              url,
               destination,
               () => resolve(destination),
               error => reject(`Error code: ${error.code}. (${error.source})`),
             );
           }),
-        );
+        )
+        .catch(friendlyErrorMessage("Your device doesn't have an active internet connection, so we weren't able to fetch your protocol at this time. Connect to a network, and try again."));
```

Comment:
Should this be "active network connection", since some protocol URLs could be local?